File: ToDoModel/Common Functions/GetFields.vb
Imports Microsoft.Office.Interop
Imports UtilitiesVB

Public Module GetFields
    Public Function CustomFieldID_GetValue(objItem As Object, ByVal UserDefinedFieldName As String) As String
        Dim OlMail As Outlook.MailItem
        Dim OlTask As Outlook.TaskItem
        Dim OlAppt As Outlook.AppointmentItem
        Dim objProperty As Outlook.UserProperty


        If TypeOf objItem Is Outlook.MailItem Then
            OlMail = objItem
            objProperty = OlMail.UserProperties.Find(UserDefinedFieldName)

        ElseIf TypeOf objItem Is Outlook.TaskItem Then
            OlTask = objItem
            objProperty = OlTask.UserProperties.Find(UserDefinedFieldName)
        ElseIf TypeOf objItem Is Outlook.AppointmentItem Then
            OlAppt = objItem
            objProperty = OlAppt.UserProperties.Find(UserDefinedFieldName)
        Else
            objProperty = Nothing
            Dim unused = MsgBox("Unsupported object type")
        End If

        CustomFieldID_GetValue = If(objProperty Is Nothing,
            "",
            If(IsArray(objProperty.Value), FlattenArry(objProperty.Value), DirectCast(objProperty.Value, String)))
    End Function
End Module

File: ToDoModel/Common Functions/GetFields.cs
using System;
using Microsoft.VisualBasic;
using UtilitiesVB;

namespace ToDoModel
{

    public static class GetFields
    {
        public static string CustomFieldID_GetValue(object objItem, string UserDefinedFieldName)
        {
            string CustomFieldID_GetValueRet = default;
            Microsoft.Office.Interop.Outlook.MailItem OlMail;
            Microsoft.Office.Interop.Outlook.TaskItem OlTask;
            Microsoft.Office.Interop.Outlook.AppointmentItem OlAppt;
            Microsoft.Office.Interop.Outlook.UserProperty objProperty;


            if (objItem is Microsoft.Office.Interop.Outlook.MailItem)
            {
                OlMail = (Microsoft.Office.Interop.Outlook.MailItem)objItem;
                objProperty = OlMail.UserProperties.Find(UserDefinedFieldName);
            }

            else if (objItem is Microsoft.Office.Interop.Outlook.TaskItem)
            {
                OlTask = (Microsoft.Office.Interop.Outlook.TaskItem)objItem;
                objProperty = OlTask.UserProperties.Find(UserDefinedFieldName);
            }
            else if (objItem is Microsoft.Office.Interop.Outlook.AppointmentItem)
            {
                OlAppt = (Microsoft.Office.Interop.Outlook.AppointmentItem)objItem;
                objProperty = OlAppt.UserProperties.Find(UserDefinedFieldName);
            }
            else
            {
                objProperty = null;
                var unused = Interaction.MsgBox("Unsupported object type");
            }

            CustomFieldID_GetValueRet = objProperty is null ? "" : objProperty is Array ? FlattenArray.FlattenArry((object[])objProperty) : (string)objProperty;
            return CustomFieldID_GetValueRet;
        }
    }
}